test/e2e/stackdriver_monitoring.go

```diff
@@ -39,5 +46,148 @@ var _ = framework.KubeDescribe("Stackdriver metrics", func() {
 	})
 })
 
-func testStackdriverMonitoring() {
+var (
+	stackdriverMetrics = []string{
+		"uptime",
+		"cpu/reserved_cores",
+		"cpu/usage_time",
+		"cpu/utilization",
+		"memory/bytes_total",
+		"memory/bytes_used",
+		"memory/page_fault_count",
+		"disk/bytes_used",
+		"disk/bytes_total",
+	}
+)
+
+func testStackdriverMonitoring(c clientset.Interface) {
+	_, err := expectedPods(c)
+	time.Sleep(time.Minute * 5)
```

Comment:
Instead of just waiting for 5 minutes, you want to periodically check whether metrics are there. See for example:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/cluster-logging/utils.go#L149
or
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/pods.go#L115